### PR TITLE
Jobvite is a sync

### PR DESCRIPTION
**Pull request by mike-burns**
- JobviteConnection responds to #sync, which means it can use the
  asynchronous SyncController framework.
- Remove unused JobviteImportsController and associated trimmings.
- Remove unused app/views/net_suite_exports/_export.html.erb template.
- ImportResult is an Enumerable, which means we can skip calling #to_a.
- Move knowledge of how to combine first and last names into the models.

https://trello.com/c/fS4LrFwi

### CODE REVIEW

File: app/models/jobvite/connection.rb [hunk 1]
```diff
@@ -30,5 +30,42 @@ def disconnect
...
+
+    def sync
+      import_results.map { |result| Result.new(**result) }
+    end
+
+    private
+
+    def import_results
+      Importer.new(
+        user,
```
Comment (base10) [inline app/models/jobvite/connection.rb]: Thoughts on passing user in with a named argument?

File: spec/models/import_result_spec.rb [hunk 2]
```diff
@@ -33,19 +33,23 @@
...
       result[candidate_one] = :failure
       result[candidate_two] = :success
 
-      expect(result.to_a).to eql [{:candidate=>{:first_name=>"Mike"},
-                                   :result=>:failure},
-                                   {:candidate=>{:first_name=>"Nolan"},
-                                    :result=>:success}]
+      actual = []
+      result = described_class.new(attribute_mapper_double)
+      result.each { |c_r| actual << c_r }
```
Comment (base10) [inline spec/models/import_result_spec.rb]: Thoughts on expanding `c_r` to `candidate` or similar?

File: app/models/import_result.rb [hunk 3]
```diff
@@ -15,9 +17,9 @@ def to_s(format = "%{candidate}: %{result}\n")
     end.join
   end
 
-  def to_a
-    results.map do |candidate, result|
-      { candidate: candidate, result: result }
+  def each(&block)
```
Comment (base10) [inline app/models/import_result.rb]: This looks like a good learning opportunity for me. What's the benefit of defining `each` here vs. `to_a`? Getting this to behave more like an Enumerable?

**Comment by base10** [comment]
The removal of `app/views/net_suite_exports/_export.html.erb` looks fine since NetSuite exports now run through the SyncsController.

**Comment by base10** [comment]
A few questions/suggestions, but I like where this is leading.

File: app/models/import_result.rb (hunk 3 above)
Comment (mike-burns) [inline app/models/import_result.rb]: Yes; the contract for an Enumerable is that the object must define `#each`. Additionally, if you want to use `#sort`, `#max`, or `#min`, the object must define `#<=>`.

By making this Enumerable we can skip calling `#to_a` explicitly before calling e.g. `#select` or `#each`.

File: app/models/jobvite/connection.rb (hunk 1 above)
Comment (mike-burns) [inline app/models/jobvite/connection.rb]: Good thinking.

File: spec/models/import_result_spec.rb (hunk 2 above)
Comment (mike-burns) [inline spec/models/import_result_spec.rb]: Thanks for the name suggestion; it's much better than `c_r`.

**Comment by mike-burns** [comment]
Is this ready to merge?

**Comment by base10** [comment]
Let me look through the latest changes quick.

**Comment by base10** [comment]
Left a couple of :abc: comments on a506bd6, otherwise LGTM.

**Comment by base10** [comment]
@mike-burns Oh, hey the build's failing, that'd be good to address :D

**Pull request merged**